### DISCUSSION TITLE
Bypass iPXE menu depending of mac address

### DISCUSSION THREAD

**Discussion by themaluxis** [ipxe/ipxe#619] [Q&A]
Hey guys :)

I currently have a PXE Server used to manage some of my machines.
I've made a PXE Menu but for some usage I would like to bypass menu selection
is that possible to configure something to  do 
if macadress = XXXXX or ipaddress=XXXXX
then execute item2 ?

Thanks for your help

here is an exemple of my current usage: 
`#!ipxe

set menu-timeout 60000
set submenu-timeout ${menu-timeout}
isset ${menu-default} || set menu-default exit
set server_ip 192.168.1.2

menu
item --gap --           -------------    TOOLS	----------------
item item1            	MEMTEST
item --gap --           -------------	DISTRIB	----------------
item item2      	    UBUNTU 
item item3		        MENU ITEM
item item4		        MENU ITEM
item shell              Shell iPXE
item exit              	Exit

choose --timeout ${menu-timeout} --default ${menu-default} target && goto ${target}

:item1
#Paramètres de démarrage pour item 1
kernel http://${server_ip}/tftpboot/memtest.gz
boot

:item2
boot

:item3
boot

:item4
boot

:shell
shell

:exit
exit`

**Comment by NiKiZe** [comment]
You can use iseq and check netX/Mac ?

**Reply by NiKiZe** [reply]
Please be sure to use netX and not net0 or net1, unless you know for sure that it will always be the first NIC.

But the better approach is to send the Mac to a server and let that generate the script.

**Reply by themaluxis** [reply]
What kind of server do you mean to put between ?

**Comment by themaluxis** [comment]
Thanks it work great, is that possible to add some syntax on ipxe script file like : 
`cat file.txt | while read line || [[ -n $line ]];
do
   iseq ${net0/mac:hexhyp} $line && set menu-default item5 
done`

**Reply by stappersg** [reply]
Please be less greedy,  _give_ besides only _take_.
Share your working configuration with your audience here.

Untested, please test and report back
```
#!ipxe
    ...
set menu-default itemDefault
iseq ${net0/mac:hex}  52:54:00:12:34:56 && set menu-default itemA
iseq ${net0/mac:hex}  52:54:00:12:ac:dc && set menu-default itemA
iseq ${net0/mac:hex}  52:54:00:ca:fe:53 && set menu-default itemA
   ...
```

And the
```
while read mac
do
   echo iseq \${net0/mac:hex} $mac && menu-default itemA
done < mac_addresses.text
```
can (and should) be done out side  iPXE.    (Please let go the wish / demand that iPXE should grow full shell functionality)

**Reply by themaluxis** [reply]
Hey stappersg, thanks for your help.
Unfortunately I can't share my plain config as it contain some confidential infos
This is working : iseq ${netX/mac:hexhyp} 0c-42-a1-23-33-fe && set menu-default item5 && set menu-timeout 5000 || set menu-default exit

Now looking for a script to easy update menu assigned to mac

**Comment by stappersg** [comment]
On Thu, Mar 17, 2022 at 03:02:59PM -0700, Maxence wrote:
> Hey stappersg, ....

That message has many transmission errors.
Please send a better version.


And regarding "What kind of server do you mean to put between ? "

Server with functionality that solves your problem.

In case you don't understand your problem:  It is still your problem.

**Reply by NiKiZe** [reply]
If you can't stop posting offensive comments, please don't post at all.
